File: src/main/java/com/hirehub/SpringServer/Entity/Referral.java
package com.hirehub.SpringServer.Entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

@Entity
@Table(name = "referrals")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Referral {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long referralId;

    @ManyToOne
    @JoinColumn(name = "employeeId")
    private Employee employee;

    @ManyToOne
    @JoinColumn(name = "referredCandidateId")
    private Candidate referredCandidate;

    @ManyToOne
    @JoinColumn(name = "jobId")
    private JobPost job;

    private Timestamp referralDate;
}

File: DotNetServer/Models/Entities/Referral.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HireHubServer.Models.Entities;

public class Referral
{
  [Key]
  public long ReferralId { get; set; }

  public long EmployeeId { get; set; }
  public long ReferredCandidateId { get; set; }
  public long JobId { get; set; }

  public DateTime ReferralDate { get; set; }

  // Navigation properties
  [ForeignKey(nameof(EmployeeId))]
  public Employee Employee { get; set; } = null!;

  [ForeignKey(nameof(ReferredCandidateId))]
  public Candidate ReferredCandidate { get; set; } = null!;

  [ForeignKey(nameof(JobId))]
  public JobPost Job { get; set; } = null!;
}
